### ISSUE TITLE
Qwen3ASR: reduce greedy decode host/GPU sync overhead

### ISSUE THREAD

**Issue by hhh2210** [ivan-digital/qwen3-asr-swift#229]
## Context

I come from an LLM/inference background and have been especially interested in improving local Mac inference efficiency. While integrating `Qwen3ASRModel` into a downstream macOS transcription app, I noticed that hot ASR inference on Apple Silicon was leaving noticeable GPU headroom during greedy decode. Activity Monitor showed GPU utilization around 50-60%, and the wall-clock profile looked consistent with a per-token host/GPU synchronization bubble rather than audio decoding.

This made me look at the path through the lens of LLM serving systems like vLLM and SGLang. I am not trying to port their full scheduler/batching stack here, but the same general idea applies: avoid unnecessary host synchronization and overlap CPU bookkeeping with already-enqueued device work where possible.

The current greedy path eventually chooses each token with:

```swift
argMax(logits, axis: -1).squeezed().item(Int32.self)
```

That `.item(Int32.self)` forces the GPU to finish the current decoder step and copy a scalar token id back to the CPU before Swift can build and enqueue the next step.

## Hypothesis

For the default greedy path, the next-token MLXArray can stay on-device long enough to stage the following decoder step first. Using `asyncEval` to kick off the next argmax/cache update before synchronizing the current token should overlap some CPU bookkeeping with the next GPU step.

This is not expected to be a large vLLM-style batching win, because the 1.7B 8-bit decoder is still mostly weight-bandwidth bound. But it should remove a small, repeatable host sync bubble without changing public API or non-greedy decoding behavior.

## Local experiment

I implemented a local branch that:

- routes only the default greedy path (`repetitionPenalty == 1`, `noRepeatNgramSize == 0`, `temperature == 0`) through a double-buffered `asyncEval` loop
- leaves the CPU-side repetition/no-repeat/sampling path unchanged
- keeps argmax dtype aligned with the legacy `Int32` token input path
- preserves the legacy append-then-EOS-break behavior

## Local benchmark

Downstream benchmark harness:

- model: `aufklarer/Qwen3-ASR-1.7B-MLX-8bit`
- audio: local lecture `.m4a`
- chunks: 4 x 15s = 60s audio per iteration
- iterations: 5, hot mean over iter 2-5
- runner: Xcode test path from a macOS app, so MLX metallib is available

| run | branch | hot wall | RTF | relative speed |
| --- | --- | ---: | ---: | ---: |
| baseline | `main` | 7.406s | 0.1234 | 1.000x |
| optimized | asyncEval branch | 6.838s | 0.1140 | **1.083x** |
| delta | — | **-0.568s** | **-0.0094** | **+8.3%** |

Per-chunk speedups were 1.075x-1.087x. This is a modest but clear improvement for a tiny API-neutral change.

Correctness check: 4 chunks produced matching text signatures across baseline and optimized runs in the same process lineage, and each branch was deterministic across 5 hot iterations. I did see one cross-process MLX nondeterminism case while investigating, so I avoided over-claiming absolute process-to-process bit identity.

## Future optimization space

This PR is meant as a small, low-risk first step rather than the end of the performance work. From an LLM serving perspective, the larger wins would likely come from infrastructure-level changes that better match ideas from vLLM / SGLang:

- **Static / continuous chunk batching**: offline ASR already has independent chunks, so batching multiple chunks through the decoder could amortize weight loads across several token streams. This is the most likely path to multi-x throughput improvements.
- **Speculative decoding**: a smaller Qwen3-ASR model could draft several tokens, then the 1.7B model could verify them in a short batched forward pass. ASR may be a good fit because the conditional entropy is relatively low.
- **Prefix / prompt KV reuse**: the repeated chat/template prefix could be cached across chunks for a smaller but composable win.

I kept the current PR deliberately narrow because it preserves public API and non-greedy behavior, and because it gives a measurable improvement without introducing a scheduler or multi-model state management. I would be very happy to discuss whether any of the larger infra directions fit the maintainers' roadmap.

## Testing caveat

Running `swift test` directly in this repo compiles, but MLX-backed tests on my machine hit:

```text
MLX error: Failed to load the default metallib. library not found
```

So the performance/correctness validation above was done through the downstream Xcode app test bundle where the MLX metallib is present. I am happy to adapt the benchmark or add a smaller repo-local regression test if you prefer a different validation shape.

## PR

@ivan-digital I have a small PR ready with the code change and would love feedback on whether this is an acceptable direction for Qwen3-ASR greedy decoding.
